### PR TITLE
fix(utils/normalizeFallback): fallback options behaviour (`options.fallback`)

### LINKED ISSUE #144: Need for `utils/normalizeFallback`
I think `utils/normalizeFallback` is not needed and we should revert the logic in `src/index.js` back to just `fallback.call(this, src)` since we need to pass all options anyways as it seems e.g

**`webpack.config.js`**
```js
{
  loader; 'url-loader',
  options: {
      limit: 1,
      name: '',
      fallback: 'x-loader',
      any: '',
      option: '',
      you: '',
      want: '',
  }
}
```

should be sufficient

Since https://github.com/webpack-contrib/url-loader/pull/139 all options are again passed to the fallback loader (default && custom), so in case I don't misunderstand the logic in `utils/normalizeFallback` this change basically neglects all extra 'work' we are currently doing there. 

Someone please correct me if I'm wrong 😛

### LINKED ISSUE #146: Issue with default fallback options
The following working config in 1.0.1:
```
{
	test: /\.(png|jpg|gif|svg)$/,
	loader: 'url-loader',
	options: {
		limit: 2048,
		name: 'img/[name].[hash:8].[ext]'
	}
}
```
does not work with version 1.1.0, i.e., the 'name' option is not passed to the default fallback 'file-loader'.
Is this on purpose or is it a bug?

### PR DESCRIPTION
**Pull request by evilebottnawi**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [x] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

fixes #144 fixes #146

### Breaking Changes

no

### Additional Info

Refactor tests + more tests

### CODE REVIEW

File: test/__snapshots__/fallback-option.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`fallback option {Object} 1`] = `"module.exports={\\"limit\\":-9007199254740991,\\"name\\":\\"fallback-[hash].[ext]\\",\\"unknown\\":\\"fallback-other-value\\"}"`;
+
+exports[`fallback option {String} (with query) 1`] = `"module.exports={\\"limit\\":-9007199254740991,\\"name\\":\\"fallback-[hash].[ext]\\",\\"unknown\\":\\"fallback-value\\"}"`;
+
+exports[`fallback option {String} 1`] = `"module.exports={\\"limit\\":-9007199254740991,\\"name\\":\\"[name].[hash].[ext]\\",\\"unknown\\":\\"value\\"}"`;
+
```
Comment (evilebottnawi) [inline test/__snapshots__/fallback-option.test.js.snap]: What we do:
1. Get original options from `url-loader`.
2. Get fallback options
3. Merge their : `options = Object.assign({}, originalOptions, fallbackOptions);`
4. delete `options.fallback`.

**Comment by michael-ciniawsky** [comment]
@Pimm @insin A second pair of eyes welcome please review when time :)

**Comment by evilebottnawi** [comment]
@michael-ciniawsky I think to pass all the options is the right solution. Also we provide way to rewrite `name` for fallback loader

**Comment by michael-ciniawsky** [comment]
Agreed I just thought passing only the custom options defined `options.fallback.options` was an initial requirement for custom fallbacks (I might be totally wrong here)

**Comment by evilebottnawi** [comment]
@michael-ciniawsky let's merge this and release new version

**Pull request merged**